### DISCUSSION TITLE
Why doesn't wallet_switchEthereumChain work?

### DISCUSSION THREAD

**Discussion by heebaebang** [WalletConnect/walletconnect-monorepo#2893] [Developer Support] [answered]
## Why doesn't `wallet_switchEthereumChain` work?
I successfully connected my wallet through the MetaMask app and when I call wallet_switchEthereumChain to change the chain from 1 (mainnet) to 80001 (mumbai) after connecting the wallet, I get the following message.
```
Failed to switch to chain 'eip155:80001'. The chain is not approved or the wallet does not support 'wallet_switchEthereumChain' method.
```
WalletConnect options
```
chains: [1],
optionalChains:[80001],
rpcMap: {
  1: "https://cloudflare-eth.com",
  80001: "https://rpc-mumbai.maticvigil.com",
}
```

For now, here are the versions I'm using
- Android Metamask 7.1.0
- `"@web3-react/walletconnect-v2": "^8.3.7"` with `"@walletconnect/ethereum-provider": "^2.8.6"`.

This worked fine in v1, so why am I getting an error? Did I miss something?

**Comment by boidushya** [comment]
Hey!
Thanks for bringing this up. We have published a version with chain switching support, MetaMask is yet to implement it. Is there any other wallet you can test with and manage to reproduce the same error?

**Reply by heebaebang** [reply]
Thanks for the response, is this issue specific to Metamask? I'm getting the same error message in the Rainbow(1.8.22) app as well.
Can you let me know which wallet you have fixed the issue for now?

**Reply by boidushya** [reply]
does passing both chains under `chains` as opposed to `optionalChains` still result in an error for you through Rainbow?
And what happens on using Rainbow? 
Meanwhile, ill create a minimal reproduction for the same and get back to you. 
Another thing worth nothing, have you passed `wallet_switchEthereumChain` under `methods` through EthereumProvider?

**Reply by heebaebang** [reply]
does passing both chains under chains as opposed to optionalChains still result in an error for you through Rainbow?
-> Attempting to connect a wallet with the Metamask app fails with an `unsupported chains` error.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/121843393/ba7c2f70-ee05-454f-901b-ede5222b387b)

And what happens on using Rainbow?
-> Unable to connect wallet with the same error as metamask.

Another thing worth nothing, have you passed wallet_switchEthereumChain under methods through EthereumProvider?
-> We use the same logic to connect wallets and switch networks via the metamask web extension with no problems.
```
    await provider.request({
      method: "wallet_switchEthereumChain",
      params: [{ chainId: hexChainId }],
    });
```

**Reply by boidushya** [reply]
Hey! An update, we are testing it internally and getting it fixed as soon as possible. We'll keep you posted about the status

**Reply by heebaebang** [reply]
This seems to be very closely related to the issue in the [link](https://github.com/MetaMask/metamask-mobile/issues/6655).

**Comment by boidushya** [comment, marked as answer]
Update (12/07/2023) :
This issue has been fixed on our end and we are waiting for a new release build from MetaMask

**Reply by heebaebang** [reply]
I'll check it out, thanks!
